services/ui_backend_service/api/ws.py

```diff
@@ -154,22 +164,25 @@ async def subscribe_to(self, ws, uuid: str, resource: str, since: int):
                 )
 
     async def unsubscribe_from(self, ws, uuid: str = None):
+        if ws not in self._subscriptions:
+            return
         if uuid:
-            self.subscriptions = list(
-                filter(lambda s: uuid != s.uuid or ws != s.ws, self.subscriptions))
+            self._subscriptions[ws] = list(
+                filter(lambda s: uuid != s.uuid or ws != s.ws, self._subscriptions[ws]))
```

Comment:
is the `ws != s.ws` check necessary anymore as the subscription is keyed per websocket?